core/appcontainer/MessageModel.js

```diff
@@ -44,6 +56,13 @@ export class MessageModel {
         this.onCancel = config.onCancel;
         this.result = new Promise(resolve => this._resolver = resolve);
 
+        // Extract properties from input
+        if (isPlainObject(this.input)) {
+            const {value, rules} = this.input;
```

Comment:
Seems like it would be more straighforward and cleaner to just create the form model here in this conditional, no?  Most messages won't need it.  Then we, also don't have to change anything with the FormModel API.